mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -56,6 +61,18 @@ sync_repo() {
 
     git checkout ${currBranch}
 
+    # we must reset Godeps.json to what it looked like BEFORE the last vendor sync so that any
+    # new Godep.json changes from k8s.io/kubernetes will apply cleanly.  Since its always auto-generated
+    # it doesn't matter that we're removing it
+    lastResyncCommit=$(git rev-list -n 1 --grep "sync: resync vendor folder" HEAD)
+    cleanGodepJsonCommit=$(git rev-list -n 1 ${lastResyncCommit}^)
+    git checkout ${cleanGodepJsonCommit} Godeps/Godeps.json
+    if git diff --cached --exit-code &>/dev/null; then
+        echo "no need to reset Godeps.json!"
+    else
+        git -c user.name="Kubernetes Publisher" -c user.email="k8s-publish-robot@users.noreply.github.com" commit -m "sync: reset Godeps.json" -- Godeps/Godeps.json
+    fi
+
```

Comment:
We could drop the complete deduplicate logic by checking before the cherry-pick whether the commit touches `Godeps/Godeps.json`. If it does (and is the first such commits in the loop), do the reset first.